atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -0,0 +1,159 @@
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, remote_source_content_manifest,
+                 destdir=IMAGE_BUILD_INFO_DIR):
+        """
+        :param tasker: ContainerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param remote_source_content_manifest: Cachito ICM
+        :param destdir: image path to carry content_manifests data dir
+        """
+        super(AddImageContentManifestPlugin, self).__init__(tasker, workflow)
+        self.icm_dir = os.path.join(destdir, 'content_manifests')
+        self.icm = remote_source_content_manifest
+        self.dfp = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        labels = Labels(self.dfp.labels)
+        _, image_name = labels.get_name_and_value(Labels.LABEL_TYPE_COMPONENT)
+        _, image_version = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
+        _, image_release = labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)
+        self.icm_file_name = '{}-{}-{}.json'.format(image_name, image_version, image_release)
+        self.out_file_path = os.path.join(self.workflow.builder.df_dir,
+                                          self.icm_file_name)
+        # Default layer index is 1, because base and 'FROM scratch' images *always* have 2 layers
+        self.layer_index = 1
+        self.content_sets = []
+
+    def _populate_content_sets(self):
+        """
+        Get the list of the current platform's content_sets from
+        'content_sets.yml' in dist-git, and set `self.content_sets` to same.
+        """
+        current_platform = self.workflow.user_params['platform']
+        workdir = self.workflow.builder.df_dir
+        fpath = os.path.join(workdir, REPO_CONTENT_SETS_CONFIG)
+        if os.path.exists(fpath):
+            content_sets_yml = read_yaml_from_file_path(fpath, 'schemas/content_sets.json') or {}
+            if current_platform in content_sets_yml:
+                self.content_sets = content_sets_yml[current_platform]
+        self.log.debug('Output content_sets: %s', self.content_sets)
+
+    def _set_layer_index(self):
+        if not base_image_is_scratch(self.dfp.baseimage):
+            inspect = self.workflow.builder.base_image_inspect
+            layer_index = len(inspect[INSPECT_ROOTFS][INSPECT_ROOTFS_LAYERS])
+            self.layer_index = layer_index
+
+    def _update_icm_data(self):
+        # Inject the content_sets data into the ICM JSON object
+        self.icm['content_sets'] = self.content_sets
+
+        # Inject the current image layer index number into the ICM JSON object metadata
+        self.icm['metadata']['image_layer_index'] = self.layer_index
+
+        # Convert dict -> str
+        icm_json = json.dumps(self.icm, indent=4)
+
+        # Validate the ICM with the ICM JSON Schema
+        read_yaml(icm_json, 'schemas/content_manifest.json')
+
+        self.log.debug('Output ICM: %s', icm_json)
+
+    def _write_json_file(self):
```

Comment:
`AddContentSetsPlugin` has same method. Reuse the code?

Reply:
see deprecation comments above

Reply:
@athos-ribeiro Thanks for reminding.

@ben-alkov Please ignore my comments that are not valid due to deprecate `add_content_sets`.